lib/response/set/setRequestCycle.js

```diff
@@ -12,17 +12,18 @@ var MaxRetryExceededError = require("./../../errors/MaxRetryExceededError");
  */
 module.exports = function setRequestCycle(model, observer, groups,
                                           isJSONGraph, isProgressive, count) {
+    var requestedAndOptimizedPaths = setGroupsIntoCache(model, groups);
+    var optimizedPaths = requestedAndOptimizedPaths.optimizedPaths;
+    var requestedPaths = requestedAndOptimizedPaths.requestedPaths;
```

Comment:
Something to consider which could be a little scary: we're going from conditionally setting groups into cache, to always setting. It might be that the shape of what is set makes this inconsequential, but could be worth a close look before you merge.

Reply:
Talked it out with Jafar. It seems logical to add any potential results returned from server into cache, even if it was on the last retried request.
Good catch!